dbt/include/trino/macros/adapters.sql

```diff
@@ -91,11 +91,33 @@
 
 {% macro trino__create_table_as(temporary, relation, sql) -%}
   {%- set _properties = config.get('properties') -%}
-  create table {{ relation }}
+
+  {%- set contract_config = config.get('contract') -%}
+  {%- if contract_config.enforced -%}
+
+  create table
+    {{ relation }}
+    {{ get_table_columns_and_constraints() }}
+    {{ get_assert_columns_equivalent(sql) }}
+    {%- set sql = get_select_subquery(sql) %}
```

Comment:
Please create an issue to investigate type specification in CTAS in Trino. We should validate if this is supported by the SQL spec or not because this way won't allow to hot swap the table using `CREATE OR REPLACE`.

Reply:
Also I would prefer this to be implemented with DESCRIBE OUTPUT instead of splitting into a CT and INSERT.